src/internal/hooks/use-performance-marks/index.ts

```diff
@@ -1,20 +1,44 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useEffect } from 'react';
-import { useUniqueId } from './use-unique-id';
-import { useEffectOnUpdate } from './use-effect-on-update';
-import { useModalContext } from '../context/modal-context';
+import { useEffect, useRef } from 'react';
+import { useIdFallback } from '../use-unique-id';
+import { useEffectOnUpdate } from '../use-effect-on-update';
+import { useModalContext } from '../../context/modal-context';
 
+/*
+This hook allows setting an HTML attribute after the first render, without rerendering the component.
+*/
+function useAttributeWithoutRerendering(elementRef: React.RefObject<HTMLElement>, value: string) {
+  const attributeName = 'data-analytics-performance-mark';
+
+  const attributeValueRef = useRef<string | undefined>();
```

Comment:
Why is this ref initially empty and does not use the `value` argument?

Reply:
If the ref were set to the attribute value initially, the attribute value would be included in the first render, which means that it would be present in server-side-rendered markup (and cause a hydration mismatch). By setting it only immediately after the first render, the attribute is present on any render except the first one.

Reply:
Got it, so we apply `useEffect` as "client-side only" technique. I think it would be useful to add a comment with this info. Leaving dependencies is fine, because it makes the eslint rule happy